### PR TITLE
Update Go examples to use $fn.Output style of invokes

### PR DESCRIPTION
**Pull request by t0yv0**
Context: pulumi/pulumi#7948

Updates Go examples to use $fnOutput feature.

The following examples cannot update yet because pulumi-azure-native Go SDK opts out of the feature via a flag:

   - azure-go-aks
   - azure-go-aks-helm
   - azure-go-appservice-docker

They do look like they would benefit.

The following examples have LookupAmiOutput that I think we should not update:

-  testing-unit-go
-  aws-go-webserver

Stats on total examples eyeballed: 

```
Go   35/35
```

### CODE REVIEW

File: classic-azure-go-webserver-component/webserver.go [hunk 1]
```diff
@@ -110,17 +110,10 @@ func NewWebserver(ctx *pulumi.Context, name string, args *WebserverArgs, opts ..
...
-		resourceGroupName := args[2].(string)
-		ip, err := network.GetPublicIP(ctx, &network.GetPublicIPArgs{
-			Name:              name,
-			ResourceGroupName: resourceGroupName,
-		})
-		if err != nil {
-			return "", err
-		}
-		return ip.IpAddress, nil
+	return ws.VM.ID().ApplyT(func(id pulumi.ID) pulumi.StringOutput {
```
[the remark this answers is not in the document]
Comment (pgavlin) [reply classic-azure-go-webserver-component/webserver.go:113]: I'm not sure which I like less--`pulumi.All` or `ApplyT(...) pulumi.Output` 😬

[the remark this answers is not in the document]
Comment (t0yv0) [reply classic-azure-go-webserver-component/webserver.go:113]: Purely aesthetic concerns? I'm open to revering. Go examples do not seem to benefit as much from the new feature, but I needed examples, so erred on the side of updating.

**Comment by t0yv0** [comment]
Not clear what's going on with CI, trying one more time.

**Comment by t0yv0** [comment]
I'm just going to close this PR. https://github.com/pulumi/examples/issues/1115 filed to follow up updating Azure Go examples that truly benefit from the feature. This is blocked on Azure Native landing the feature.

**Pull request closed without merging**